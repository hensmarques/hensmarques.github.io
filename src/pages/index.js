import React from "react"

import Layout from "../components/layout"
import About from "../components/about"
import Timeline from "../components/timeline"
import Portfolio from "../components/portfolio"
import Footer from "../components/footer"

import SEO from "../components/seo"

const IndexPage = () => (
    <Layout>
        <SEO title="Home" /> 
        <About></About>
        <Timeline></Timeline>
        <div className="mb-24"></div>
        <Portfolio></Portfolio>
        <Footer></Footer>
    </Layout>
)
    
export default IndexPage
    