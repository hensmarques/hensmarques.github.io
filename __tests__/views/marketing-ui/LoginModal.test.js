import React from 'react';
import TextInput from '../../../src/elements/form-controls/TextInput';
import { LoginModal } from '../../../src/views/marketing-ui/LoginModal';
import { authStatus, authError } from '../../../src/reducers';

describe('<LoginModal/>', () => {
  describe('Login Form', () => {
    const login = jest.fn();
    const onClose = jest.fn();
    const authNone = jest.fn();
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <LoginModal
          login={login}
          authStatus={authStatus.none}
          authNone={authNone}
        />
      );
    });

    it('renders and matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('dispatches login action from form submit', () => {
      const loginCredentials = {
        password: 'bar'
      };
      const instance = wrapper.instance();
      expect(login).not.toHaveBeenCalled();

      wrapper
        .find('TextInput')
        .at(0)
        .simulate('change', { target: { value: loginCredentials.username } });
      wrapper
        .find('TextInput')
        .at(1)
        .simulate('change', { target: { value: loginCredentials.password } });
      instance.submit();
      expect(login).toHaveBeenCalledWith(
        wrapper.state().formFields.username,
        wrapper.state().formFields.password,
        '',
        ''
      );
    });
  });

  describe('Verification Code Form', () => {
    const login = jest.fn();
    const onClose = jest.fn();
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <LoginModal
          login={login}
          onClose={onClose}
          authStatus={authStatus.failed}
          authError={{ error: authError.verificationCode }}
        />
      );
    });

    it('renders and matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders input for verfication code on verification code authError', () => {
      wrapper.setProps({
        authStatus: authStatus.failed,
        authError: { error: authError.verificationCode }
      });
      expect(wrapper.exists('div.verification-code-form')).toBe(true);
    });

    it('dispatches login action from form submit', () => {
      const loginCredentials = {
        username: 'foo',
        password: 'bar',
        verificationCode: '123456'
      };
      const instance = wrapper.instance();
      const { verificationCode, ...prevState } = loginCredentials;
      wrapper.setState({ formFields: prevState });

      wrapper.find('TextInput').simulate('change', {
        target: { value: verificationCode }
      });
      instance.submit();
      expect(login).toHaveBeenCalledWith(
        wrapper.state().formFields.username,
        wrapper.state().formFields.password,
        wrapper.state().formFields.verificationCode,
        wrapper.state().formFields.confirmationCode,
      );
    });

    it('dispatches login action from verification code form submit', () => {
      const loginCredentials = {
        username: 'foo',
        password: 'bar',
        verificationCode: '123456',
        confirmationCode: '123456',
      };
      const instance = wrapper.instance();
      wrapper.setProps({
        authStatus: authStatus.failed,
        authError: { error: authError.verificationCode }
      });
      wrapper.setState({
        formFields: {
          username: loginCredentials.username,
          password: loginCredentials.password
        }
      });

      wrapper.find('TextInput').simulate('change', {
        target: { value: loginCredentials.verificationCode }
      });
      expect(login).toHaveBeenCalledWith(
        wrapper.state().formFields.username,
        wrapper.state().formFields.password,
        wrapper.state().formFields.verificationCode,
        wrapper.state().formFields.confirmationCode,
      );
    });
  });

  describe('Authentication functionality', () => {
    it('renders an error message if there is an authError for an invalid verification code', () => {
      const mockProps = {
        authStatus: authStatus.failed,
        authError: {
          error: authError.invalidVerificationCode
        }
      };
      const wrapper = shallow(<LoginModal {...mockProps} />);
      expect(wrapper.exists('div.error')).toBe(true);
    });

    it('redirects to home page when user authenticates', () => {
      const testAuthStatus = authStatus.success;
      const wrapper = shallow(<LoginModal authStatus={testAuthStatus} />);
      expect(wrapper.find('Modal').exists()).toBe(false);
      expect(wrapper.find('Redirect').exists()).toBe(true);
      expect(wrapper.find('Redirect').props().to).toBe('/home');
    });
  });
});
