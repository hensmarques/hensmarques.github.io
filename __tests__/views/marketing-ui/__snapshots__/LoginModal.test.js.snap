// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginModal/> Login Form renders and matches the snapshot 1`] = `
<div
  className="login-modal"
>
  <Connect(Modal)
    className="marketing-modal"
  >
    <Form
      onSubmit={[Function]}
    >
      <div
        className="form-instructions"
        key="0"
      />
      <div
        className="login-form"
        key="1"
      >
        <TextInput
          className=""
          disabled={true}
          errorMessages={Array []}
          hideInput={false}
          name="Email"
          onChange={[Function]}
          placeholder="Email"
          placeholderParam={Object {}}
          translation="Email"
          type="text"
          value=""
          writeOnce={false}
        />
        <TextInput
          className=""
          disabled={true}
          errorMessages={Array []}
          hideInput={true}
          name="Password"
          onChange={[Function]}
          placeholder="Password"
          placeholderParam={Object {}}
          translation="Password"
          type="text"
          value=""
          writeOnce={false}
        />
      </div>
      <div
        className="form-message"
        key="2"
      >
        <p
          className="forgot-password"
        >
          Forgot your password?
        </p>
        <p>
          <span
            className="link"
            onClick={[Function]}
          >
            Click here to reset.
          </span>
        </p>
      </div>
      <Submit
        className=""
        formValid=""
        onSubmit={[Function]}
        value="LOGIN"
      />
    </Form>
  </Connect(Modal)>
  <Link
    onClick={[Function]}
    replace={false}
    to="/home"
  >
    <img
      className="xIcon"
      src="/images/icons/xThin.svg"
    />
  </Link>
</div>
`;

exports[`<LoginModal/> Verification Code Form renders and matches the snapshot 1`] = `
<div
  className="login-modal"
>
  <Connect(Modal)
    className="marketing-modal"
  >
    <Form
      onSubmit={[Function]}
    >
      <div
        className="form-instructions"
        key="0"
      >
        Enter verification code from Google Authenticator to continue logging in.
      </div>
      <div
        className="verification-code-form"
        key="1"
      >
        <TextInput
          className="error"
          disabled={true}
          errorMessages={Array []}
          hideInput={false}
          name="Verification Code"
          onChange={[Function]}
          placeholder="Verification Code"
          placeholderParam={Object {}}
          translation="Verification Code"
          type="text"
          value=""
          writeOnce={false}
        />
      </div>
      <div
        className="form-message"
        key="2"
      >
        <p
          className="forgot-password"
        >
          Forgot your password?
        </p>
        <p>
          <span
            className="link"
            onClick={[Function]}
          >
            Click here to reset.
          </span>
        </p>
      </div>
      <Submit
        className=""
        formValid=""
        onSubmit={[Function]}
        value="LOGIN"
      />
    </Form>
  </Connect(Modal)>
  <Link
    onClick={[Function]}
    replace={false}
    to="/home"
  >
    <img
      className="xIcon"
      src="/images/icons/xThin.svg"
    />
  </Link>
</div>
`;
